### ISSUE TITLE
"The type "type[FunctionType]" is not generic and not indexable" false positive on new type alias syntax

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#738]
```py
type ClassMethod[T, **P, R] = FunctionType[Concatenate[type[T], P], R]
```

**Comment by DetachHead** [comment]
it also reports `valid-type` with th following message:
```
Type parameters for "FunctionType" requires __future__.annotations or quoted types
```
even when `__future__.annotations` is imported

**Comment by KotlinIsland** [comment]
- #739
